bin/resmoke-tags.js

```diff
@@ -4,26 +4,35 @@ const path = require("path");
 const util = require("util");
 
 const program = require("commander");
-const CLIEngine = require('eslint').CLIEngine;
+const { ESLint } = require('eslint');
 
 function localeCompare(str1, str2) {
     return str1.localeCompare(str2);
 }
 
-function lint(files, ruleOptions) {
-    const cli = new CLIEngine({
+async function lint(files, ruleOptions) {
+    const eslint = new ESLint({
         // We don't use the top-level .eslintrc.yml file but we still enable ES6 features and
         // MongoDB global variables ourselves.
         useEslintrc: false,
-        envs: ["es6", "mongo"],
-
-        plugins: ["mongodb-server"],
-        rules: {"mongodb-server/resmoke-tags": ruleOptions},
+        overrideConfig: {
+            env: {
+                "es6": true,
```

Comment:
Let's change this line to `"es2022": true` match https://github.com/mongodb/mongo/blob/2cf2693a7846bb49b9e819c8be4863e096c85ebe/.eslintrc.yml#L2. Ditto line 15.